frontend/lib/src/components/widgets/DataFrame/columns/cells/LinkCell.tsx

```diff
@@ -0,0 +1,174 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from "react"
+
+import {
+  CustomCell,
+  CustomRenderer,
+  getMiddleCenterBias,
+  measureTextCached,
+  blend,
+  GridCellKind,
+} from "@glideapps/glide-data-grid"
+import UriOverlayEditor from "./UriOverlayEditor"
+import { isNullOrUndefined } from "@streamlit/lib/src/util/utils"
+
+const UNDERLINE_OFFSET = 5
+
+export interface LinkCellProps {
+  readonly kind: "link-cell"
+  readonly href?: string | null
+  readonly displayText?: string | null
+}
+
+export type LinkCell = CustomCell<LinkCellProps>
+
+function onClickSelect(
+  e: Parameters<NonNullable<CustomRenderer<LinkCell>["onSelect"]>>[0]
+): string | null | undefined {
+  const canvas = document.createElement("canvas")
+  const ctx = canvas.getContext("2d", { alpha: false })
+  if (ctx === null) return
+
+  const { posX: hoverX, bounds: rect, cell, theme } = e
+  const font = `${theme.baseFontStyle} ${theme.fontFamily}`
+  ctx.font = font
+
+  const { href, displayText } = cell.data
+
+  const rectHoverX = rect.x + hoverX
+
+  const textWidth = ctx.measureText((displayText || href) ?? "").width
+  const textStart = rect.x + theme.cellHorizontalPadding
+
+  const isHovered =
+    rectHoverX > textStart && rectHoverX < textStart + textWidth
+
+  if (isHovered) {
+    return href
+  }
+
+  return undefined
+}
+
+export const linkCellRenderer: CustomRenderer<LinkCell> = {
+  draw: (args, cell) => {
+    const { ctx, rect, theme, hoverX = -100, highlighted } = args
+    const { href, displayText } = cell.data
+    if (isNullOrUndefined(href)) return
+
+    const displayValue = displayText || href
+
+    const xPad = theme.cellHorizontalPadding
+
+    const drawX = rect.x + xPad
+
+    const rectHoverX = rect.x + hoverX
+
+    const font = `${theme.baseFontStyle} ${theme.fontFamily}`
+    ctx.font = font
+    const middleCenterBias = getMiddleCenterBias(ctx, font)
+    const drawY = rect.y + rect.height / 2 + middleCenterBias
+
+    const isHovered = rectHoverX > rect.x && rectHoverX < rect.x + rect.width
+
+    if (isHovered) {
+      // draw the underline only when the cell is hovered
+      const metrics = measureTextCached(displayValue, ctx, font)
+
+      ctx.moveTo(drawX, Math.floor(drawY + UNDERLINE_OFFSET) + 0.5)
+      ctx.lineTo(
+        drawX + metrics.width,
+        Math.floor(drawY + UNDERLINE_OFFSET) + 0.5
+      )
+
+      ctx.strokeStyle = theme.linkColor
+      ctx.stroke()
+
+      ctx.fillStyle = highlighted
+        ? blend(theme.accentLight, theme.bgCell)
+        : theme.bgCell
+      ctx.fillText(displayValue, drawX - 1, drawY)
+      ctx.fillText(displayValue, drawX + 1, drawY)
+
+      ctx.fillText(displayValue, drawX - 2, drawY)
+      ctx.fillText(displayValue, drawX + 2, drawY)
```

Comment:
I think we can remove those:

```suggestion
```

This is applied to create these spacings in the underline of the link. But without the spacings it is actually looking a lot more like the HTML version:

Canvas link with these spacings:

<img width="214" alt="image" src="https://github.com/streamlit/streamlit/assets/2852129/dd2f0dbc-5331-4695-909e-68faa20c8ed5">

Canvas link without these spacings

<img width="214" alt="image" src="https://github.com/streamlit/streamlit/assets/2852129/836f745a-f05d-4ef0-a8c5-f6243299266a">


Link in overlay editor (this is just the basic `a` HTML version): 

<img width="214" alt="image" src="https://github.com/streamlit/streamlit/assets/2852129/fc55b1ca-9188-4f31-80ff-7df47a2751ba">

So the version without the spacings is actually closer to the HTML version.

Reply:
Sounds good! I wasn't sure if having the spacings was preferred or not, but if not then less code is obviously better.

Reply:
Yeah, I'm fine without the spacings.. especially since it's the same as the HTML default.